### PR TITLE
E_CLASSROOM-286 (Dependent #288) [FE/BE] Categories sort by column

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-286

### Definition of Done
- [x] User should be able to sort by id or name

## Related PR:
- https://github.com/framgia/sph-classroom-els-fe/pull/131
#### Main note
- http://localhost:82/api/v1/admin/categories-data
- Make sure to add these following query parameters when testing in Postman;
> sortBy ---> can only accept these string values: Name, Category, and ID [The first letters should be capitalized, while the ID should be all caps]
> sortDirection ----> can only accept these string values: asc or desc [should be small letters]
> page ---> you can set this to 1 for the pagination
- For the list to go back to it's original order, just leave the sortBy and sortDirection blank, it still needs to be in the parameters but with no values.

#### Pages Affected
- N/A

#### Scenarios/ Test Cases
- [x] it should sort by id or name

### Screenshots
categoryadmin1

### CODE REVIEW

File: app/Traits/SortCategories.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            case 'Name': 
+                $sortBy = 'name';
+                break;
+            default:
+                $sortBy = 'categories.'.strtolower($query['sortBy']);
+        }
+        
+        return $categories->orderBy($sortBy, $query['sortDirection'])->paginate(12);
+    }
+}
```
Comment (karlolee50) [inline app/Traits/SortCategories.php]: add extra space

File: app/Traits/SortCategories.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use App\Models\Category;
+
+trait SortCategories
+{
+    protected function sort($query, $categories)
+    {
+        switch($query['sortBy']){
+            case 'Name': 
+                $sortBy = 'name';
+                break;
```
Comment (karlolee50) [inline app/Traits/SortCategories.php]: is this still needed? we are already using strtolower which will make all the string lowercase. can you double check?

File: routes/api.php [hunk 3]
```diff
@@ -83,5 +83,6 @@
         Route::get('/admin/nested-categories/{subcategory_id}', [CategoryController::class, 'getSubcategoryParentCategories']);
         Route::post('/admin/profile-edit', [ChangeNameEmailController::class, 'restore']);
         Route::patch('/admin/password-edit', [AdminController::class, 'updatePassword']);
+        Route::get('/admin/categories-data', [CategoryController::class, 'listOfCategories']);
```
Comment (karlolee50) [inline routes/api.php]: i think it is better for this to be /admin/categories. just update the other one to /admin/categories-data

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 2

**Comment by Teraaay** [comment]
### Approved PR #67
- [x] Able to sort by ID and Name

### Screenshot/s
> SORT CATEGORIES BY COL FINALgif

----
- Total Test time: 0.25 hr
- Total Iterations: 0
- Total Comments: 0

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**